modepy/quadrature/__init__.py

```diff
@@ -82,3 +102,83 @@ def __init__(self, quad, left, right):
         Quadrature.__init__(self,
                 left + (quad.nodes+1) / 2 * length,
                 quad.weights * half_length)
+
+
+class TensorProductQuadrature(Quadrature):
+    """
+    .. attribute:: exact_to
+
+        The tensor product quadrature is exact up to this summed polynomial degree.
```

Comment:
Not liking this much. `exact_to` is part of the `Quadrature` base interface, and this class shouldn't change its mind on what that means. What I was proposing is to document (in `Quadrature`) that exactness means exactness on P^N of that degree (in the multi-D) case.

Reply:
The docs for `Quadrature.exact_to` already mention it's about the "total degree". Isn't that sufficient to imply `P^N` even in higher dimensions?

The tensor product quadrature would integrate up to a "maximum degree", right?

Reply:
> The docs for `Quadrature.exact_to` already mention it's about the "total degree"

Oof. Good point. Then most of the data returned is currently incorrect. (see [here](https://en.wikipedia.org/wiki/Degree_of_a_polynomial#Extension_to_polynomials_with_two_or_more_variables))

I propose we change that wording to P^N (giving a weaker guarantee, but one that matches the data that's actually being returned).

Reply:
How does 10a17d5 look?

Reply:
LGTM